libfwupdplugin/fu-optionrom-device.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2015 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "config.h"
+
+#include "fu-optionrom-device.h"
+
+G_DEFINE_TYPE(FuOptionromDevice, fu_optionrom_device, FU_TYPE_PCI_DEVICE)
+
+static gboolean
+fu_optionrom_device_probe(FuDevice *device, GError **error)
+{
+	FuOptionromDevice *self = FU_OPTIONROM_DEVICE(device);
+	gboolean rom_exists = FALSE;
+	g_autofree gchar *rom_fn = NULL;
+
+	/* does the device even have ROM? */
+	rom_fn = g_build_filename(fu_udev_device_get_sysfs_path(FU_UDEV_DEVICE(self)), "rom", NULL);
+	if (!fu_device_query_file_exists(device, rom_fn, &rom_exists, error))
+		return FALSE;
+	if (rom_exists)
+		fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_CAN_VERIFY_IMAGE);
```

Comment:
I'd just fail probe here if it doesn't exist.  There is no point in having such a device otherwise.

Reply:
Well I say that but then I remembered amd-gpu inherits from here and doesn't require the image verification stuff on APUs.